compiler/src/dotty/tools/dotc/printing/RefinedPrinter.scala

```diff
@@ -734,7 +734,7 @@ class RefinedPrinter(_ctx: Context) extends PlainPrinter(_ctx) {
         keywordStr("macro ") ~ toTextGlobal(call)
       case tree @ Quote(body, tags) =>
         val tagsText = (keywordStr("<") ~ toTextGlobal(tags, ", ") ~ keywordStr(">")).provided(tree.tags.nonEmpty)
-        val exprTypeText = (keywordStr("[") ~ toTextGlobal(tree.bodyType) ~ keywordStr("]")).provided(printDebug && tree.typeOpt.exists)
+        val exprTypeText = (keywordStr("[") ~ toTextGlobal(tpd.bodyType(tree.asInstanceOf[tpd.Quote])) ~ keywordStr("]")).provided(printDebug && tree.typeOpt.exists)
```

Comment:
The `provided` condition makes sure that the type exists. If it does not, the text and `tpd.bodyType` is not evaluated as it is a by-name prefix.

The other solution to this would be to create `bodyTypeOpt`.

Reply:
Ah I missed the provided, ok.